autosubmitconfigparser/config/configcommon.py

```diff
@@ -851,12 +859,12 @@ def clean_dynamic_variables(self, pattern, in_the_end=False):
 
         for key, dynamic_var in dynamic_variables_.items():
             # if not placeholder in dynamic_var[1], then it is not a dynamic variable
-            if isinstance(dynamic_var[1], list):
-                for value in dynamic_var[1]:
+            if isinstance(dynamic_var, list):
+                for value in dynamic_var:
                     if not re.search(pattern, value, flags=re.IGNORECASE):
                         dynamic_variables[key] = dynamic_var
```

Comment:
I'm quite concerned about this part because the purpose of the function is to keep the variables that match the pattern and in this part keep the last one that doesn't match :dizzy_face: 

I'll do this instead:

```suggestion
            if isinstance(dynamic_var, list):
                matching_values = []
                for value in dynamic_var:
                    if re.search(pattern, value, flags=re.IGNORECASE):
                        matching_values.append(value)
                if len(matching_values) > 0:
                    dynamic_variables[key] = matching_values
```

And it should satisfy this test:

```python
def test_clean_dynamic_variables_list(autosubmit_config: Callable) -> None:
    """Test cleaning dynamic (non-special) variables.

    Read the text in the test above (``test_clean_dynamic_variables_special_variables``),
    as that can be helpful here.

    This tests that once called with the right arguments, ``clean_dynamic_variables`` will
    leave ``as_conf.dynamic_variables`` with only dynamic variables.
    """
    in_the_end = False

    as_conf: AutosubmitConfig = autosubmit_config(expid='a000', experiment_data={})
    dynamic_variables, pattern, start_long = as_conf._initialize_variables(in_the_end=in_the_end)

    as_conf.dynamic_variables = {
        'popeye_eats': 'spinach',
        'penguin_eats': 'fish',
        'thor_eats': ['%^DYNAMIC_1%', '%^DYNAMIC_3%', '%PLACEHOLDER2%'],
        'floki_eats': '%^DYNAMIC_2%',
        'jaspion_eats': ['%PLACEHOLDER%', '%^DYNAMIC_4%']
    }

    as_conf.clean_dynamic_variables(pattern, in_the_end=in_the_end)

    assert len(as_conf.dynamic_variables) == 2
    assert 'thor_eats' in as_conf.dynamic_variables
    assert as_conf.dynamic_variables['thor_eats'] == ['%PLACEHOLDER2%']
    assert 'jaspion_eats' in as_conf.dynamic_variables
    assert as_conf.dynamic_variables['jaspion_eats'] == ['%PLACEHOLDER%']

```